pages/login.vue

```diff
@@ -1,3 +1,83 @@
+<script setup>
+    // import { generateCodeVerifier,  generateChallenge } from 'pkce-challenge'
+    import pkceChallenge from 'pkce-challenge'
+
+    const client_id = "20aa48c2719e42c0be5f3b834942f06d"
+    const scopes = []  // https://developer.spotify.com/documentation/general/guides/authorization/scopes/
+    const redirect_uri = new URL(window.location)
+    redirect_uri.search = ""  // clear search stuff
+    const auth_token = localStorage.getItem('auth-token')
+
+    async function logout(){
+        localStorage.removeItem('auth-token')
+        navigateTo('/')
+    }
+
+    async function login(){
+        const pkce = pkceChallenge()
+
+        localStorage.setItem('code-verifier', pkce.code_verifier)
+
+        const url = new URL('/authorize', 'https://accounts.spotify.com/')
+        url.searchParams.append('client_id', client_id)
+        url.searchParams.append('response_type', 'code')
+        url.searchParams.append('redirect_uri', redirect_uri)
+        url.searchParams.append('scopes', scopes.join(" "))
+        url.searchParams.append('code_challenge_method', 'S256')
+        url.searchParams.append('code_challenge', pkce.code_challenge)
+        navigateTo(url.toString(), {external: true})
+    }
+
+    async function loadToken(){
+        const code_verifier = localStorage.getItem('code-verifier')
+
+        const url = new URL('/api/token', 'https://accounts.spotify.com/')
+        const form = new FormData()
+        form.append('grant_type', "authorization_code")
+        form.append('code', pkce)
+        form.append('redirect_uri', redirect_uri)
+        form.append('client_id', client_id)
+        form.append('code_verifier', code_verifier)
+        
+        const response = await fetch(url, {
+            method: "POST",
+            body: new URLSearchParams(form),  // don't ask. i have no idea
```

Comment:
Also das können wir doch nicht in den master branch lassen, das geht bestimme ein bisschen sauberer. Sollte der body nicht einfach die Daten in JSON format enthalten?